runtime/src/current_sandbox.c

```diff
@@ -67,13 +68,27 @@ current_sandbox_start(void)
 	char *error_message = "";
 
 	sandbox_initialize_stdio(sandbox);
+	struct module * next_module = sandbox->module->next_module;
 
-	sandbox_open_http(sandbox);
+	if (sandbox->request_from_outside || next_module == NULL) {
```

Comment:
Will add comment for this line. The first and last sandbox in the chain should use epoll to monitor the fd because they needs to read and write from/to this fd. `sandbox->request_from_outside` indicates the first sandbox in the chain, `next_module == NULL` indicates the last sandbox in the chain.